pkg/scheduler/util/utils.go

```diff
@@ -119,6 +122,33 @@ func PatchPodStatus(ctx context.Context, cs kubernetes.Interface, old *v1.Pod, n
 	return err
 }
 
+// PrepareAndDeletePod prepares and deletes the given <pod> from API server
+func PrepareAndDeletePod(ctx context.Context, cs kubernetes.Interface, pod *v1.Pod, podConditionType v1.PodConditionType) error {
```

Comment:
Since multiple binaries will be using these functions, they should go in staging/src/k8s.io/component-helpers